api/api/clubs/tests.py

```diff
@@ -191,3 +192,178 @@ def testUpdateClubs_positionHolderRequest_updatesSuccessful(position):
+
+
+@pytest.mark.django_db
+def testCreateClubRegistrationRequest_ClubDoesNotExist_returnBadRequest(client, test_user):
+    # given
+    client.force_authenticate(test_user)
+
+    # when
+    response = client.post('/clubs/19999/register/', {
+        'fee_submitted': True,
+        'status': 'Pending',
+        'remark': 'Some text'
+    })
+
+    # then
+    assert response.status_code == status.HTTP_400_BAD_REQUEST
+    assert response.data['detail'] == 'club does not exist'
+
+
+@pytest.mark.django_db
+def testCreateClubRegistrationRequest_validRequest_returnHTTP201(client, test_user, test_club):
+    # given
+    client.force_authenticate(test_user)
+
+    # when
+    response = client.post(f'/clubs/{test_club.id}/register/', {
+        'fee_submitted': True,
+        'status': 'Pending',
+        'remark': 'Some text'
+    })
+
+    # then
+    assert response.status_code == status.HTTP_201_CREATED
+    assert response.data == {'id': 1, 'user': 1, 'club': 1,
+                             'fee_submitted': True, 'status': 'Pending', 'remark': 'Some text'}
+
+
+@pytest.mark.django_db
+def testUpdateClubRegistrationRequest_nonAdminRequest_throwsForbidden(client, test_user,
+                                                                      test_registration_request):
+    # given
+    client.force_authenticate(test_user)
+
+    # when
+    response = client.put(f'/clubs/registration/{test_registration_request.id}/', {
+        'fee_submitted': True,
+        'remark': 'Late fees',
+        'status': 'Approved',
+    })
+
+    # then
+    assert response.status_code, status.HTTP_403_FORBIDDEN
+
+
+@ pytest.mark.django_db
+def testUpdateClubRegistration_adminReq_invalidUpdateId_returnNotFound(client):
+    # given
+    user = User.objects.create(
+        email='admin@user.com', first_name='admin', last_name='user', is_staff=True)
+    client.force_authenticate(user)
+
+    # when
+    response = client.put('/clubs/registration/99/', {
+        'fee_submitted': True,
+        'remark': 'Late fees',
+        'status': 'Approved',
+    })
+
+    # then
+    assert response.status_code == status.HTTP_404_NOT_FOUND
+
+
+@pytest.mark.django_db
+def testUpdateClubRegistrationRequest_adminRequest_updatesSuccessful(client,
+                                                                     test_registration_request):
+    # given
+    user = User.objects.create(
+        email='admin@user.com', first_name='admin', last_name='user', is_staff=True)
+    client.force_authenticate(user)
+
+    # when
+    response = client.put(f'/clubs/registration/{test_registration_request.id}/', {
+        'fee_submitted': True,
+        'remark': 'Late fees',
+        'status': 'Approved',
+    })
+
+    # then
+    updated_registration_request = ClubRegistrationRequest.objects.get(
+        pk=test_registration_request.id)
+    assert response.status_code == status.HTTP_200_OK
+    assert response.data, {'id': test_registration_request.id,
```

Comment:
Shouldn't this be ==? instead of the comma?
other places too.

Reply:
Typo